proposals/NNNN-atf-improvment-proposal.md

```diff
@@ -0,0 +1,79 @@
+# Improve Automated test framework
+
+* Proposal: [SDL-NNNN](NNNN-atf-improvement-proposal.md)
+* Author: [Alexander Kutsan](https://github.com/smartdevicelink)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / ATF]
+
+## Introduction
+
+The proposal is to make ATF more flexible, salable and easy to use and integrate with popular CI systems and to analyze and full of all required information. 
+Also some functionality of ATF should be moved to scripts common files due to it's dependency from SDL version and features.
+
+## Motivation
+
+Current ATF requires and unfortunately allows changes during testing specific SDL scenarios. 
+Also it requires additional environment preparation and management during testing test sets
+and integrating ATF to popular CI systems.
+Need to make ATF more flexible and easy to use. And extend it scaling possibilities.
+A lot of common functionality should be moved from ATF to scripts common. 
+
+## Proposed solution
+
+There is 3 different layers of using ATF:
+
+1. Local executing and debugging concrete scenario even on local SDL build and script
+2. Local test set execution, Logs, reports, traffic analyses. Set set can be executed on remote SDL and on local SDL build
+3. CI execution of test set, collection Logs, reports, traffic logs for further analysis 
+
+ATF should be easy to use as for executing and debugging one concrete scenario on local workstation even with customly modified SDL.
+
+ATF architecture should be modified to support all this 3 layers of execution. Also ATF should support testing SDL on remote workstation/target. 
+
+Automatic creating connection and mobile session with SDL, and HMI - SDL connection should be removed from ATF.
```

Comment:
Automatic connection creation or
Automatic connection of connection and mobile session